### PR TITLE
Fix west-of-UTC time zone bug

### LINKED ISSUE #192: get_local_time_zone() broken for time zones with minutes west of UT
The timezone functions are not working properly for non-standard timezones behind UTC that have either:
- zero hours but non-zero minutes (e.g. -00:30)
- minutes that aren't 0 or 30 (e.g. -02:15)

#### Steps to reproduce

(Note: the `TZ` env var +/- convention is backwards, so that Eastern Standard Time is written `TZ=EST+05:00` instead of UTC-05:00)

```
$ TZ=XXX+02:15 python
>>> from metomi.isodatetime import timezone
>>> timezone.get_local_time_zone()
(-2, 45)  # should be (-2, 15) or (-2, -15)
>>> timezone.get_local_time_zone_format()
'-0245'  # should be '-0215'
```

```
$ TZ=XXX+00:30 python
>>> from metomi.isodatetime import timezone
>>> timezone.get_local_time_zone()
(0, 30)  # should be (0, -30)
>>> timezone.get_local_time_zone_format()
'+0030'  # should be '-0030'
```

#### Additional context

If the timezone is east of UTC, it works ok

```
$ TZ=XXX-02:15 python
>>> from metomi.isodatetime import timezone
>>> timezone.get_local_time_zone()
(2, 15)
>>> timezone.get_local_time_zone_format()
'+0215'
```

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #192

Fixes a bug where `metomi.isodatetime.timezone` would return incorrect results for certain non-standard/unusual system time zones. Also, `get_local_time_zone()` now always returns negative minutes for time zones behind UTC. 

This PR also drops support for Python 3.5, and makes `-m 'not slow'` the default for pytest.

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -14,10 +12,16 @@
...
-# Some tests fail when running in a different time zone (e.g. Pacific/Auckland)
-markers = slow
-testpaths = metomi/isodatetime/tests
+addopts =
+    -v
+    -s
+    -ra
+    --color=auto
+    --doctest-glob='*.md'
+    -m 'not slow'
```
Comment (kinow) [inline pytest.ini:23]: One job in GH Actions timed out after 45 minutes. Maybe there's something missing here?

**Review by kinow — APPROVED** [review]
Code and tests look good! Leaving one comment about a job that got stuck, but not related to the west-of-UTC time zone bug, not a blocker I think. Thanks!

```bash
(venv) kinow@ranma:~/Development/python/workspace/isodatetime$ TZ=EST+05:00 python
Python 3.8.3 (default, May 19 2020, 18:47:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from metomi.isodatetime import timezone
>>> timezone.get_local_time_zone()
(-5, 0)
>>> 
```

File: pytest.ini (hunk 1 above)
Comment (MetRonnie) [reply pytest.ini:23]: I'm pretty sure it's just the coverage run being slow

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -50,7 +50,7 @@ jobs:
         env:
           TZ: ${{ matrix.tz }}
         run: |
-          PYTEST_ARGS=('-n' '2')
+          PYTEST_ARGS=(-n 2 -m 'slow or not slow')
```
Comment (oliver-sanders) [inline .github/workflows/test.yml:53]: Isn't `slow or not slow` equivalent to not specifying any marks?

**Review by oliver-sanders — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (MetRonnie) [reply .github/workflows/test.yml:53]: No because of https://github.com/metomi/isodatetime/blob/19929352802aa4c4fe9a72a0475c410775dc58bc/pytest.ini#L22

**Comment by MetRonnie** [comment]
Merging with 2 approvals

**Pull request merged**